### PR TITLE
Add Embedded Swagger UI Viewer

### PR DESCRIPTION
**Pull request by irishshagua**
* JavaFX based Swagger UI View
    * Open swagger files with split view
    * Enable displaying / hiding / refreshing swagger view

Reference open issue:
https://github.com/zalando/intellij-swagger/issues/105

### CODE REVIEW

**Comment by semonte** [comment]
Thank you for the PR, this will be great! I am unable to use the split view. Is there some button I need to click, toggle some views? Here is a screenshot:

image

**Comment by irishshagua** [comment]
Hey @semonte 
To test I ran
`gradle runIdea`

And then I get a view like below:
image

And then clicking on the swagger icon gives:
image

How are you running the plugin? My first time playing with the intellij SDK so could well have messed up something... :-/

**Comment by irishshagua** [comment]
Hey @semonte 

I tested the plugin like:
`gradle runIdea`

Then I see the below when I open a swagger file:
screen shot 2017-12-22 at 12 56 07

And when I click the swagger icon I get:
screen shot 2017-12-22 at 12 56 25

**Comment by semonte** [comment]
Ok got it, it is working. I had to run `./gradlew clean runIdea` and reopen the file in the editor. I will review the code next week, but thank you already in advance for doing this.

File: src/main/java/org/zalando/intellij/swagger/service/SwaggerFileService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+public class SwaggerFileService {
+
+    private Map<PsiFile, Path> convertedSwaggerDocuments = new HashMap<>();
+
+    private SwaggerUiCreator swaggerUiCreator = new SwaggerUiCreator(new FileContentManipulator());
+    private FileDetector fileDetector = new FileDetector();
+
+
+    public Optional<Path> convertSwaggerToHtml(@NotNull PsiFile swaggerFile) {
```
Comment (semonte) [inline src/main/java/org/zalando/intellij/swagger/service/SwaggerFileService.java]: I would find this method more readable if it was split to smaller chunks like:

```java
final Path swaggerUiDirectory = updateOrCreateSwaggerUiFiles(swaggerFile);
publishChanges(swaggerUiDirectory);
        
return swaggerUiDirectory;
```

**Comment by semonte** [comment]
Works and looks good 👍 . I would still ask you to add a `final` keyword wherever possible, that has been the coding practice for this project. I had only one comment regarding a long method.

**Comment by irishshagua** [comment]
Cool, will try and submit a revision tonight. Just getting the kids to bed at the mo... 😂

**Pull request merged**